cdm/src/main/java/thredds/crawlabledataset/s3/CachingThreddsS3Client.java

```diff
@@ -98,6 +108,41 @@ public ObjectListing listObjects(S3URI s3uri) {
         return objectListing.orNull();
     }
 
+    @Override
+    public ThreddsS3Metadata getMetadata(S3URI s3uri) {
+        Optional<ThreddsS3Metadata> metadata = metadataCache.getIfPresent(s3uri);
+        if (metadata == null) {
```

Comment:
No - getIfPresent returns null if the object does not exist on the cache

I can have a null object stored on the cache using Optional (actually just copied this from unidata guys caching code)